runtime/integration-tests/src/generic/cases/liquidity_pools.rs

```diff
@@ -3942,9 +3935,24 @@ mod development {
 						DEFAULT_DOMAIN_ADDRESS_MOONBEAM,
 						decrease_msg_pool_swap_amount
 					));
+					assert!(frame_system::Pallet::<T>::events().iter().any(|e| {
```

Comment:
Super NIT. There is a `find_event()` in `generic/utils/mod.rs`

Reply:
I don't find this API to be more helpful here because all foreign investment integration tests operate in a single closure of `env` such that for multiple of these events it is better to directly look for the exact same one as we are asserting at least two values here, not only a single one.

But I will definitely remember this for the next integration tests.

Reply:
Thanks for the explanation. Next time, I touch the integration tests, I will re-check whether it makes sense to switch to that API :)